sdk/eventhubs/azure_messaging_eventhubs/src/lib.rs

```diff
@@ -19,779 +16,18 @@ pub mod error;
-
-        assert_eq!(event_data.body().unwrap(), &body);
-    }
-
-    #[test]
-    fn test_event_data_builder_with_content_type() {
-        let content_type = "application/json";
-        let event_data = EventData::builder()
-            .with_content_type(content_type.to_string())
-            .build();
-
-        assert_eq!(event_data.content_type(), Some(content_type));
-    }
-
-    #[test]
-    fn test_event_data_builder_with_correlation_id() {
-        let correlation_id = MessageId::String("correlation-id".to_string());
-        let event_data = EventData::builder()
-            .with_correlation_id(correlation_id.clone())
-            .build();
-
-        assert_eq!(event_data.correlation_id(), Some(&correlation_id));
-    }
-
-    #[test]
-    fn test_event_data_builder_with_message_id() {
-        let message_id = MessageId::String("message-id".to_string());
-        let event_data = EventData::builder()
-            .with_message_id(message_id.clone())
-            .build();
-
-        assert_eq!(event_data.message_id(), Some(&message_id));
-    }
-
-    #[test]
-    fn test_event_data_builder_add_property() {
-        let key = "key".to_string();
-        let value: AmqpValue = "value".into();
-        let event_data = EventData::builder()
-            .add_property(key.clone(), value.clone())
-            .build();
-
-        assert_eq!(event_data.properties().unwrap().get(&key), Some(&value));
-    }
-
-    #[test]
-    fn test_event_data_builder_build() {
-        let body = vec![1, 2, 3];
-        let content_type = "application/json";
-        let correlation_id = MessageId::String("correlation-id".to_string());
-        let message_id = MessageId::String("message-id".to_string());
-        let key = "key".to_string();
-        let value: AmqpValue = "value".into();
-
-        let event_data = EventData::builder()
-            .with_body(body.clone())
-            .with_content_type(content_type.to_string())
-            .with_correlation_id(correlation_id.clone())
-            .with_message_id(message_id.clone())
-            .add_property(key.clone(), value.clone())
-            .build();
-
-        assert_eq!(event_data.body().unwrap(), &body);
-        assert_eq!(event_data.content_type(), Some(content_type));
-        assert_eq!(event_data.correlation_id(), Some(&correlation_id));
-        assert_eq!(event_data.message_id(), Some(&message_id));
-        assert_eq!(event_data.properties().unwrap().get(&key), Some(&value));
-    }
-}
+/// Types sent to and received from the EventHubs service.
+pub mod models;
+
+pub use producer::batch::*;
+pub use producer::ProducerClient;
+pub use producer::ProducerClientOptions;
+pub use producer::SendEventOptions;
+pub use producer::SendMessageOptions;
+pub use producer::SubmitBatchOptions;
+
+pub use consumer::ConsumerClient;
+pub use consumer::ConsumerClientOptions;
+pub use consumer::OpenReceiverOptions;
+pub use consumer::StartLocation;
+pub use consumer::StartPosition;
```

Comment:
So you're exporting `consumer` and `producer` modules, and then these top-level clients similar to our "normal" Azure SDK guidelines currently? What are the `StartLocation` and `StartPosition` types? Sound like models.

Reply:
`StartLocation` and `StartPosition` are types referenced by the `ProducerClientOptions` and are not model types (we use them to format strings which are sent to the service). 

As I recall it, services are expected to expose their service client objects in their top-level client, even if they're located in a `clients` (or `producer` and `consumer`) module. I believe that also applies to all the `Options` types needed to create the client. 

I don't have any preferences here.